### PR TITLE
Legacy redirect

### PR DESCRIPTION
**Pull request by CommanderStorm**
This Route either:
- Not found
- Permanent Redirect
- image

The Gameplan is the following:
image

### CODE REVIEW

**Review by octycs — APPROVED** [review]
This seems like a good temporary feature. I've just added a redirect for buildings using the "@<building id>" format used by the old roomfinder.

File: server/src/core/legacy_redirect.rs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    }
+}
+
+fn extract_redirect_base(key: TypeAndKey) -> String {
+    match key._type.as_str() {
+        "root" => "".to_string(),
+        "joined_building" => format!("building/{}", key.key),
+        "virtual_room" => format!("room/{}", key.key),
+        "area" => format!("site/{}", key.key), // Currently also "site", maybe "group"? TODO
+        res => format!("{}/{}", res, key.key),
```
[the remark this answers is not in the document]
Comment (CommanderStorm) [reply server/src/core/legacy_redirect.rs]: No, that is not what I meant.
I have updated the code to make my intent more clear.

[the remark this answers is not in the document]
Comment (CommanderStorm) [reply server/src/core/legacy_redirect.rs]: Is change I added more clear?

[the remark this answers is not in the document]
Comment (octycs) [reply server/src/core/legacy_redirect.rs]: Ah yes, now I understand

**Review by octycs — APPROVED** [review]

**Pull request merged**